src/lib/opts.rs

```diff
@@ -249,6 +249,65 @@ impl Opts {
         }
         Ok(output_types_to_write)
     }
+
```

Comment:
The following method is needed since index hopping metrics and some command line option validation requires the length of every sample barcode segment to be known.